numpy/core/src/private/ufunc_override.c

```diff
@@ -34,47 +38,33 @@ has_non_default_array_ufunc(PyObject *obj)
 
     /* Fast return for ndarray */
     if ((PyObject *)Py_TYPE(obj) == ndarray) {
-        return 0;
+        return NULL;
     }
     /* does the class define __array_ufunc__? */
-    cls_array_ufunc = PyArray_GetAttrString_SuppressException(
-                          (PyObject *)Py_TYPE(obj), "__array_ufunc__");
```

Comment:
It's worth noting that before and after this patch, `__array_ufunc__` would fail on old-style classes.